precompiles/erc20/integration_test.go

```diff
@@ -659,20 +687,18 @@ var _ = Describe("ERC20 Extension -", func() {
 
 				// Approve allowance
 				txArgs, approveArgs := s.getTxAndCallArgs(callType, contractData, auth.ApproveMethod, grantee.Addr, tokenCoins[0].Amount.BigInt())
-				approveArgs.MethodName = auth.ApproveMethod
-				approveArgs.Args = []interface{}{grantee.Addr, tokenCoins[0].Amount.BigInt()}
 
 				approveCheck := passCheck.WithExpEvents(auth.EventTypeApproval)
 
 				_, _, err = s.factory.CallContractAndCheckLogs(granter.Priv, txArgs, approveArgs, approveCheck)
 				Expect(err).ToNot(HaveOccurred(), "unexpected result calling contract")
 
 				// Check allowance contains both spend limits
-				s.ExpectSendAuthzForContract(callType, contractData, grantee.Addr, granter.Addr, bondCoins.Add(tokenCoins...))
+				s.expectSendAuthz(grantee.AccAddr, granter.AccAddr, bondCoins.Add(tokenCoins...))
 			},
 				Entry(" - direct call", directCall),
-				// NOTE: we are not passing the contract call here because this test case only covers direct calls
-				//
+				// FIXME: This fails because it's creating the different authorization (see above)
+				Entry(" - through contract", contractCall),
```

Comment:
pending authorization fix on our end (authorization is created with the contract address as owner instead of the granter)

Reply:
This behavior aligns with the ERC20 implementation, but differs from our other EVM extensions - for compatibility with the ERC20 standard, we keep it as it is.